OpenMarket/OpenMarket/Application/Presentation/ViewController/MarketProductsViewController.swift

```diff
@@ -0,0 +1,193 @@
+//
+//  OpenMarket - MarketProductsViewController.swift
+//  Created by 데릭, 케이, 수꿍. 
+//  Copyright © yagom. All rights reserved.
+//
+
+import UIKit
+
+final class MarketProductsViewController: UIViewController {
+    fileprivate enum Section {
+        case main
+    }
+    
+    private let segmentedControl: UISegmentedControl = {
+        let control = UISegmentedControl(items: ["LIST", "GRID"])
+        control.translatesAutoresizingMaskIntoConstraints = false
+        return control
+    }()
+    
+    private var shouldHideListView: Bool? {
+        didSet {
+            guard let shouldHideListView = self.shouldHideListView else {
+                return
+            }
+            
+            self.listCollectionView?.isHidden = shouldHideListView
+            self.gridCollectionView?.isHidden = !self.listCollectionView.isHidden
+        }
+    }
+    
+    private var productsModel: [ProductEntity] = []
+    
+    private var listCollectionView: UICollectionView!
+    private var listDataSource: UICollectionViewDiffableDataSource<Section, ProductEntity>!
+    
+    private var gridCollectionView: UICollectionView!
+    private var gridDataSource: UICollectionViewDiffableDataSource<Section, ProductEntity>!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        view.backgroundColor = .white
+        
+        setNavigationItems()
+        fetchData()
+    }
+}
+
+
+private extension MarketProductsViewController {
+    func fetchData() {
+        let url = "https://market-training.yagom-academy.kr/api/products?page_no=1&items_per_page=50"
+        
+        let openMarket = NetworkProvider(session: URLSession.shared)
+        openMarket.requestAndDecode(url: url, dataType: ProductList.self) { result in
+            switch result {
+            case .success(let productList):
+                productList.pages.forEach { product in
+                    let item = ProductEntity(thumbnailImage: product.thumbnailImage!, name: product.name, currency: product.currency, originalPrice: product.price, discountedPrice: product.bargainPrice, stock: product.stock)
```

Comment:
```suggestion
                    let item = ProductEntity(
                       thumbnailImage: product.thumbnailImage!, 
                       name: product.name, 
                       currency: product.currency, 
                       originalPrice: product.price, 
                       discountedPrice: product.bargainPrice, 
                       stock: product.stock
                    )
```
너무 길면 띄워써주는 게 훨씬 좋아요. 여러 스타일 가이드를 참고해보세요~

Reply:
image 그것도 서버로부터 받은 이미지를 forced unwrapping 하면 크래시 많이 나고 사용자 경험이 급격히 떨어집니다.
고치고 앱 제출하는데 시간도 매우 오래걸려요 
uiimage는 옵셔널 타입이어도 uimageview.image 에 할당가능하니 옵셔널타입 그대로 entity에 넘기시는게 나을듯 합니다.